ambry-cloud/src/main/java/com.github.ambry.cloud/azure/CosmosDataAccessor.java

```diff
@@ -0,0 +1,197 @@
+/**
+ * Copyright 2019 LinkedIn Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ */
+package com.github.ambry.cloud.azure;
+
```

Comment:
We should have unit tests for CosmosDataAccessor. I see that most of the methods are only called from AzureCloudDestination right now, and there are already tests for AzureCloudDestination. But still some tests required for CosmosDataAccessor (e.g, for retywiththrottling, queryMetadata etc). Also there are methods in AzureCloudDestination like  getDeadBlobs, updateMetadata etc that might require more testing.

Reply:
I will add a specific test to exercise the retryWithThrottling logic.  The other methods are already tested by AzureCloudDestinationTest and/or AzureIntegrationTest.

Reply:
I agree that in general it's preferable to have a unit test for every production class, so I will consider refactoring the AzureCloudDestinationTest in a follow up PR.

Reply:
Created test class.